### PR TITLE
e2e(apprunner): reduce size of dependencies

### PR DESCRIPTION
**Pull request by dannyrandall**
- upgrade to go 1.19
- use stdlib http router
- migrate to aws sdk v2 to reduce dependency size
- use distroless container

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: e2e/apprunner/front-end/Dockerfile [hunk 1]
```diff
@@ -1,33 +1,15 @@
...
 
-# To make our images smaller, use alpine and copy in the service binary.
-FROM alpine:latest
-# Install certs
-RUN apk --no-cache add ca-certificates
-# Copy the binary from the builder image
-COPY --from=builder /app ./
-# Make the binary executable
-RUN chmod +x ./e2e-service
+FROM gcr.io/distroless/static
```
Comment (Lou1415926) [inline e2e/apprunner/front-end/Dockerfile:11]: interesting!
> "Distroless" images contain only your application and its runtime dependencies. They do not contain package managers, shells or any other programs you would expect to find in a standard Linux distribution.

File: e2e/apprunner/front-end/main.go [hunk 2]
```diff
@@ -40,11 +41,11 @@ func SimpleGet(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
 // This test assumes the backend app is called "back-end". The 'service-discovery' endpoint
 // of the back-end service is unreachable from the LB, so the only way to get it is
 // through service discovery. The response should be `back-end-service-discovery`
-func ServiceDiscoveryGet(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+func ServiceDiscoveryGet(w http.ResponseWriter, req *http.Request) {
 	endpoint := fmt.Sprintf("http://back-end.%s/service-discovery/", os.Getenv("COPILOT_SERVICE_DISCOVERY_ENDPOINT"))
 	resp, err := http.Get(endpoint)
 	if err != nil {
-		log.Printf("🚨 could call service discovery endpoint: err=%s\n", err)
+		log.Printf("🚨 could call service discovery endpoint: err=%s", err)
```
Comment (Lou1415926) [inline e2e/apprunner/front-end/main.go:48]: tiny q: why do we not want the new line anymore?

File: e2e/apprunner/front-end/main.go [hunk 3]
```diff
@@ -64,36 +65,43 @@ type Secret struct {
...
 		SecretId: aws.String(os.Getenv("FRONTENDCLUSTER_SECRET_ARN")),
 	})
 	if err != nil {
 		log.Printf("🚨 retrieve aurora secret: err=%s\n", err)
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
-	secretValue := aws.StringValue(out.SecretString)
+
+	secretValue := aws.ToString(out.SecretString)
```
Comment (Lou1415926) [inline e2e/apprunner/front-end/main.go:91]: oh did they rename it 👀

**Review by Lou1415926 — APPROVED** [review]

File: e2e/apprunner/front-end/main.go (hunk 2 above)
Comment (dannyrandall) [reply e2e/apprunner/front-end/main.go:48]: `log.Printf` adds a newline already! https://pkg.go.dev/log#Logger.Output

File: e2e/apprunner/front-end/main.go (hunk 3 above)
Comment (dannyrandall) [reply e2e/apprunner/front-end/main.go:91]: lol yeah, gotta save those 3 characters 💪📈

File: e2e/apprunner/back-end/go.mod [hunk 4]
```diff
@@ -1,5 +1,3 @@
 module github.com/aws/copilot-cli/e2e/appprunner/back-end
 
-go 1.14
-
-require github.com/julienschmidt/httprouter v1.3.0 // indirect
```
Comment (efekarakus) [inline e2e/apprunner/back-end/go.mod:5]: heck yea

File: e2e/apprunner/front-end/go.mod [hunk 5]
```diff
@@ -1,9 +1,23 @@
 module github.com/aws/copilot-cli/e2e/apprunner/front-end
 
-go 1.14
+go 1.19
 
 require (
-	github.com/aws/aws-sdk-go v1.43.6
-	github.com/julienschmidt/httprouter v1.3.0
+	github.com/aws/aws-sdk-go-v2 v1.16.16
```
Comment (efekarakus) [inline e2e/apprunner/front-end/go.mod:6]: omg even our codebase doesn't use v2 😂

**Review by efekarakus — APPROVED** [review]

**Pull request merged**